src/App.js

```diff
@@ -2,25 +2,43 @@ import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList"; //remember the ./ because it's in a different file
 import AddTodoForm from "./AddTodoForm";
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] =
-    useState(JSON.parse(localStorage.getItem("savedTodoList"))) ?? [];
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  // made initial state empty array to start with an empty list and simulating fetching todos asynchronously
   // the todoList is the savedTodoList
-  // parsing to make string an object
-  // ?? operator takes the left side when right side is null
+
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    new Promise((resolve, reject) => {
+      setTimeout(
+        () =>
+          resolve({
+            data: {
+              todoList: JSON.parse(localStorage.getItem("savedTodoList")),
+            },
+          }),
+        2000
+      );
+    }).then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+    // parsing to make string an object
+  }, []);
+  // mimicing asynchronous testing of an API
+  // returning the promise with the then method
+  // using state setter to update list and pass initial state from result object
+
+  useEffect(() => {
+    if (isLoading === false) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    }
   }, [todoList]);
```

Comment:
you'll need to add `isLoading` as a dependency. Otherwise this effect won't run when `isLoading` changes. If you open up your console you should see an error relating to this